onnxruntime/test/providers/cpu/tensor/unsqueeze_op_test.cc

```diff
@@ -60,7 +60,12 @@ TEST(TensorOpTest, Unsqueeze_scalar) {
     test.AddAttribute("axes", std::vector<int64_t>{-1});
     test.AddInput<float>("input", {}, std::vector<float>{1.0f});
     test.AddOutput<float>("output", {1}, std::vector<float>{1.0f});
-    test.Run();
+    // TODO: ONNX shape inference thinks the output should be rank 0.
+    // ONNX spec is ambiguous: https://github.com/onnx/onnx/issues/2428.
+    // Once spec clarified, remove strict_shape_type_inference override.
```

Comment:
Actually ONNX shape inference for UnSqueeze-11 or later version can handle scalar well. UnSqueeze-1 cannot handle scalar here because it cannot handle negative axes for like case here. Such a negative axes support was introduced after opset version 11: https://github.com/onnx/onnx/pull/2281 Thus, I would suggest the test here (using negative axes) should specify to use opset_version >= 11 to solve the ONNX shape inference error.

Reply:
Done, thanks for looking into it!